pages/search_page.py

```diff
@@ -10,16 +10,19 @@ class SearchPage(BasePage):
     __CLOSE_BUTTON_FOR_CONFIRMATION_POPUP = (By.XPATH, '//*[@id="layer_cart"]/div[1]/div[1]/span')
     __SEARCH_RESULT_ITEM = (By.XPATH, '//*[@id="center_column"]/ul/li[1]/div')
     __ADD_BUTTON_FOR_SEARCH_RESULT_ITEM = (By.XPATH, '//*[@id="center_column"]/ul/li[1]/div/div[2]/div[2]')
-    __5TH_PRODUCT = (By.XPATH, '//*[@id="center_column"]/ul/li[5]')
+
+    # dynamic xpaths
+    __PRODUCT = (By.XPATH, '//*[@id="center_column"]/ul/li[{index}]')
+
+    def check_if_search_block_contains_element(self, expected_product_quantity: int) -> bool:
+        element_locator = (By.XPATH, self.__PRODUCT.format(index=expected_product_quantity))
```

Comment:
you can use `format` method on strings. right now you are trying to call it on a tuple:
`    __PRODUCT = (By.XPATH, '//*[@id="center_column"]/ul/li[{index}]')`